src/ssh/azext_ssh/bastion_utils.py

```diff
@@ -0,0 +1,315 @@
+
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+# pylint: disable=protected-access
+
+import platform
+import subprocess
+import tempfile
+import threading
+import time
+import json
+import uuid
+
+import requests
+from enum import Enum
+from azure.cli.core import azclierror 
+
+from azure.cli.core.commands.client_factory import get_subscription_id
+from knack.log import get_logger
+from azure.cli.core.aaz import *
+from knack.prompting import prompt_y_n
+
+
+from .aaz.latest.network.bastion import Create 
+from .vendored_sdks.resourcegraph._resource_graph_client import ResourceGraphClient
+from . import resource_graph_utils
+from . import ssh_utils
+
+logger = get_logger(__name__)
+
+
+class BastionSku(Enum):
+
+    Basic = "Basic"
+    Standard = "Standard"
+    Developer = "Developer"
+    QuickConnect = "QuickConnect"
+
+# ============================= Put Call to create a bastion Developer Sku============================= #
+def create_bastion(cmd, op_info, vnet_id):
+    try:
+        bastion_creator = Create(cli_ctx=cmd.cli_ctx)
+        bastion_args = {
+            "location": op_info.location,
+            "name": f"{op_info.resource_group_name}-vnet-bastion",
+            "resource_group": op_info.resource_group_name,
+            "sku": "Developer",
+            "virtual_network": {
+                "id": vnet_id
+            },
+            "ip_configurations": [],
+            "scale_units": 2,
+            "tags": {}
+        }
+        result = bastion_creator(command_args=bastion_args)
+         
+        op_info.bastion_name = f"{op_info.resource_group_name}-vnet-bastion"
+    except Exception as e:
+        raise azclierror.ClientRequestError(f"Failed to create bastion information")
+
+#============================================== Get call for current bastion ====================================================#
+def show_bastion(cmd, op_info):
+    from .aaz.latest.network.bastion import Show
+    try:
+        bastion = Show(cli_ctx=cmd.cli_ctx)(command_args={
+            "resource_group": op_info.resource_group_name,
+            "name": op_info.bastion_name 
+        })
+
+        return bastion
+    except Exception:
+        raise azclierror.CLIInternalError("Failed to get bastion information. Please try again later.")
+
+
+
+# ============================= SSH to Bastion Host Logic ============================= #
+
+
+def ssh_bastion_host(cmd, op_info, delete_keys, delete_cert):
+
+    bastion = show_bastion(cmd, op_info)
+
+    if bastion['sku']['name'] not in [BastionSku.Developer.value, BastionSku.QuickConnect.value]:
+        raise azclierror.InvalidArgumentValueError("SSH to Bastion host is only supported for Developer Skus.")
+
+    if op_info.port != None:
+        port = op_info.port
+        if port != 22:
+            raise azclierror.InvalidArgumentValueError("Custom Ports are not allowed for the Bastion Developer Sku. Please try again with port 22`.")
```

Comment:
I think we should call it some time after the private/ public ip are decided because in the case someone decides to use bastion after it would have to catch the same error